### DISCUSSION TITLE
Button Layout

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1590] [Proposals]
### Goal

Discutir e avaliar a possibilidade implementar o Button Layout (Button Primary Small + Button Secondary Small) na jornada de Área Aberta.

### Context

Todas as vezes que a pessoa usuária deslizar a tela para baixo e a linha de base do Callout se sobrepor a linha de base da Nav bar, uma transição deve acontecer para disponibilizar na tela, de maneira fixa como a Nav bar e Tab bar, o componente de barra responsiva com os botões de ENTRAR e CRIAR CADASTRO (50% +50%).

Ao deslizar de baixo para cima, em direção ao topo, ocorre o comportamento inverso: quando a linha da base do Callout estiver sobreposta a linha de base da Nav bar, ocorre a transição da barra responsiva para o Callout.

### Impact

Em relação ao Sistema de Design, a ideia é refletir no Figma o que já existe em código apenas deixando possível alternar a ordem dos botões, com isso, não precisaríamos criar atomicamente cada vez que esse "componente" precisar ser implementado nos fluxos.

### Discussion Questions

ND

### Do you have a visual proposal?

https://www.figma.com/file/DREYCD6OEZicpOygZ2Sg6F/Button-Layout?type=design&node-id=1%3A86&mode=design&t=344E1VjUmfEKS70T-1

**Comment by annefranca** [comment]
@yceballost

**Reply by yceballost** [reply]
Hi Anne,

This is not a request for Mística, this is possible today with Mística using [UseIsInViewport](https://mistica-web.vercel.app/?path=/story/hooks-useisinviewport--use-is-in-viewport)

an example: https://tinyurl.com/ylljlcel

This logic has to be implemented inside the product (Novum)

**Reply by annefranca** [reply]
@yceballost 

Estamos falando sobre o Button Layout, certo? A discussão é sobre a ordem dos botões (Primary + Secondary).

No Figma não temos o Button Layout, mas em código, o mesmo existe.

https://mistica-web.vercel.app/playroom/#?code=N4Igxg9gJgpiBcIA8AFATjAbgSxgdwBUIIAbAZwAIALbWAPiQHp0tdDjy6AdAOyQCEIADwoAHAIZQo2HgHMAvMACMANgC%2B3HhQoCArgBd9EHgBlxATwgGK4ktlk9N27XsPH02ALbi05iu4wyMkUACgBKCnk6CmA1DQBBMH1sYyZ%2BAyMeD29fJ2dXTIBlGEgeKB8-AJgg0IiomLi6ROTUxnS3HmLS8tzeFzaM4zNLA0004THBEQkpGQVldSdecaFNEAAaEH0qGE9qhABtEAA1bEwIAH0AOXwQAF1NvFptskOAZhUABju1IA

**Reply by yceballost** [reply]
oh, sorry, I misunderstood the request. Yes, ButtonLayout has a specific button order as you show in Playroom example. 

Why do you want to change this order?

**Reply by annefranca** [reply]
Seguindo uma lógica já existente em componentes com ação, temos a ação primária antes da ação secundária. 

No caso do exemplo, "entrar" e "criar conta".

**Reply by yceballost** [reply]
There is another discussion according to ButtonLayout, we can analyze this request together

https://github.com/Telefonica/mistica-design/discussions/680

We have some problems having these two components living together: ButtonLayout and ButtonGroup.

The reason why the buttonLayout place the primary button on the right is because in mobile this position is more affordable than the left side when the buttonLayout cover 100% of the width. This component is made mainly for using in buttonFixedFooterLayout and get this interaction more affordable as I said

On the other hand, the component used within the components is called ButtonGroup was designed to allow for multiple buttonGroups on the screen and keep visual coherence.

Anyway, we need to refactor both button layouts (ButtonLayout + ButtonGroup) to clean this mess, absolutely

**Reply by annefranca** [reply]
@yceballost com essa refatoração teríamos a possibilidade de ter flexibilidade no Button Layout. 

Como ficaria no Mística Figma? Seria um template?

No nosso caso, não conseguimos usar o Button Group porque não pega 100% da largura (50% para cada ação).

**Reply by yceballost** [reply]
> @yceballost con esta refactorización tendríamos la posibilidad de tener flexibilidad en el diseño de los botones.

It is a possibility, depending on the refactoring :D 

> ¿Cómo se vería en Mystique Figma? ¿Sería una plantilla?

Should we include in Figma the same (~) components than mistica-web. We know that there is no buttonLayout in Figma, but that's because there were talks about merging the buttonLayout and buttonGroup into a single component. And we didn't want to put it in and then take it out. It's also not a component that is used a lot, that's why we also thought about removing it. But I'm just speculating, we have to sit down and work in depth on all this.

Please, create a Jira ticket in Novo App Components to include in the roadmap. And please, prioritize if it is important for Q1 or not, we are defining the task to work on the next quarter

**Reply by annefranca** [reply]
Ticket criado :)

**Reply by annefranca** [reply]
@yceballost temos alguma atualização sobre esse ticket?

**Reply by yceballost** [reply]
We are closing IconButton and Row with iconButton  ([OBVIVO-2098](https://jira.tid.es/browse/OBVIVO-2098) & [OBVIVO-2092](https://jira.tid.es/browse/OBVIVO-2092)) and the next component prioritized is [OBVIVO-2105](https://jira.tid.es/browse/OBVIVO-2105) but I think the next week we can start with this one and ButtonLayout ([OBVIVO-2182](https://jira.tid.es/browse/OBVIVO-2182))

**Comment by yceballost** [comment]
https://github.com/Telefonica/mistica-design/issues/1674
